npk/src/manifest.rs

```diff
@@ -539,7 +680,50 @@ mounts:
     type: dev
 ";
         assert!(Manifest::from_str(manifest).is_err());
+        Ok(())
+    }
 
+    /// Overlapping mounts are invalid
+    #[test]
+    fn overlapping_mount() -> Result<()> {
+        let manifest = "name: hello\nversion: 0.0.0\ninit: /binary\nuid: 1000\ngid: 1001
+mounts:
+  /lib/overlapping:
+    type: bind
+    host: /lib
```

Comment:
would be good also to test the other conditions (Two mount paths both starting with '/', ...)